test/test_elbo.jl

```diff
@@ -36,23 +38,6 @@ function test_bvn_cov()
 end
 
 
-function test_derivative_flags()
```

Comment:
Per my other comments about the changed semantics, I would really leave this test in, and add an additional test that if calculate_gradient is false but calculate_hessian is either the combination is disallowed or you get all zeros and not garbage.